app/client/lib/localization.ts

```diff
@@ -68,14 +70,48 @@ export async function setupLocale() {
 }
 
 /**
- * Resolves the translation of the given key, using the given options.
+ * Resolves the translation of the given key using the given options.
  */
-export function t(key: string, args?: any): string {
-  if (!i18next.exists(key)) {
+export function t(key: string, args?: any, instance = i18next): string {
+  if (!instance.exists(key)) {
     const error = new Error(`Missing translation for key: ${key} and language: ${i18next.language}`);
     reportError(error);
   }
-  return i18next.t(key, args);
+  return instance.t(key, args);
+}
+
+
+/**
+ * Resolves the translation of the given key and substitutes. Supports dom elements interpolation.
+ */
+ export function domT(key: string, args?: any, instance = i18next): DomContents {
+  if (!instance.exists(key)) {
+    const error = new Error(`Missing translation for key: ${key} and language: ${i18next.language}`);
+    reportError(error);
+  }
+  // If there are any DomElements in args, handle it with missingInterpolationHandler.
+  const domElements = Object.entries(args).filter(([_, value]) => isLikeDomContents(value));
```

Comment:
If `isLikeDomContents` is inexpensive, would it be reasonable to just rename your `domT` to `t`, and current `t` to `translateWithoutInterpolation`? So translator can generally just always use the same `t` function unless there is some very specific reason not to consider interpolating?

Reply:
Yes I'm happy to switch it, I started with that idea initially, but then I had some doubts, as it won't be consistent with the server side. But I think this is what we want most of the time on client side.

Reply:
I see. Yes, I think this would be a reasonable kind of inconsistency.

Reply:
I've included simple type inference, so for plain values the result should be a string. @arnaudpeich I also included a helper method `makeT` that should be helpful and can eliminate the need to create `translate` helpers by hand.

Reply:
Looks great, thank you! I'll update our PR accordingly :)